packages/util-cadence/src/index.js

```diff
@@ -0,0 +1,47 @@
+import {invariant} from "@onflow/util-invariant"
+
+/**
+ * Parses a Cadence transaction to extract function arguments.
+ *
+ * @param {string} cadenceCode - The Cadence transaction code.
+ * @returns {Array<object>} An array of objects each containing 'name' and 'type' of each argument in the transaction.
+ * Each object has the format: { name: string, type: string }. If no valid arguments are found, or if the Cadence
+ * transaction is not properly formatted, the function will return an empty array.
+ *
+ * @example
+ * const cadenceCode = 'transaction(name: String, age: Int)'
+ * parseArguments(cadenceCode)
+ * // Returns: [{ name: 'name', type: 'String' }, { name: 'age', type: 'Int' }]
+ *
+ * @example
+ * const cadenceCode = 'transaction()'
+ * parseArguments(cadenceCode)
+ * // Returns: []
+ */
+export const parseArguments = cadenceCode => {
+  const mainFunctionIndex = cadenceCode.indexOf('pub fun main')
+  const transactionIndex = cadenceCode.indexOf('transaction')
```

Comment:
This one is a real edge case but what if I tried to send the following script or an equivalent transaction?  (I think the transaction case will be far more common than the script case)

```cadence
// pub fun main()
/* pub fun main() */
pub struct Foo {
pub fun main() {}
init() {
let bar = "pub fun main()"
}
}

pub fun main() {}
```

Any of these could throw off the parser. I know they are edge cases but I feel like it might make sense to verify that we are in the other scope and outside of a comment/string literal.

I made a JS Fiddle that avoids this, it's far from perfect but maybe some inspiration. https://jsfiddle.net/h9r8ob6x/1/